cmake/OpenCVFindLAPACK.cmake

```diff
@@ -108,11 +108,27 @@ macro(ocv_lapack_check)
 
     set(LAPACK_TRY_COMPILE_DEF "")
     if(LAPACK_IMPL STREQUAL "LAPACK/Apple") # https://github.com/opencv/opencv/issues/24660
-      set(LAPACK_TRY_COMPILE_DEF "-DACCELERATE_NEW_LAPACK")
       # TODO: Need to find a way detecting macOS / iOS versions
       #       enbale if macOS >= 13.3 or iOS >= 16.4
-      # add_compile_definitions(ACCELERATE_NEW_LAPACK)
-      # add_compile_definitions(ACCELERATE_LAPACK_ILP64)
+
+      # Get macOS version
+      execute_process(COMMAND sw_vers -productVersion
+                      OUTPUT_VARIABLE MACOS_VERSION
+                      OUTPUT_STRIP_TRAILING_WHITESPACE)
```

Comment:
This is a problem... it checks the version of macOS on the machine doing the compiling, it does not check the version of macOS that's actually running the code (at runtime). It's common for a developer to be using a new macOS but wanting his code to still run on an old macOS, and that's why these kinds of checks are inappropriate.

Instead, we must check against the '[deployment target](https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_DEPLOYMENT_TARGET.html)' chosen by the developer, set in `CMAKE_OSX_DEPLOYMENT_TARGET`.

Could we revert this merge, or correct this quickly?

Reply:
Let me take care of this. cc @asmorkalov @opencv-alalek